src/libraries/Microsoft.Extensions.Http/src/Logging/LoggingHttpMessageHandler.cs

```diff
@@ -59,16 +59,31 @@ async Task<HttpResponseMessage> Core(HttpRequestMessage request, bool useAsync,
                 // not really anything to surround.
                 _logger.LogRequestStart(request, shouldRedactHeaderValue);
                 var stopwatch = ValueStopwatch.StartNew();
-                HttpResponseMessage response = useAsync
-                    ? await base.SendAsync(request, cancellationToken).ConfigureAwait(false)
+
+                try
+                {
+                    HttpResponseMessage response = useAsync
+                        ? await base.SendAsync(request, cancellationToken).ConfigureAwait(false)
 #if NET
-                    : base.Send(request, cancellationToken);
+                        : base.Send(request, cancellationToken);
 #else
-                    : throw new NotImplementedException("Unreachable code");
+                        : throw new NotImplementedException("Unreachable code");
 #endif
-                _logger.LogRequestEnd(response, stopwatch.GetElapsedTime(), shouldRedactHeaderValue);
+                    _logger.LogRequestEnd(response, stopwatch.GetElapsedTime(), shouldRedactHeaderValue);
 
-                return response;
+                    return response;
+                }
+                catch (HttpRequestException ex)
+                {
+                    var statusCode =
+#if NET_8_OR_GREATER
+                        (int)ex.StatusCode;
+#else
+                        500;
```

Comment:
Hey @huoyaoyuan,
Thank you for the comment!
Yes, you are right:
In case if a request completes and returns failure status, it means that we will have a response log. 

The case what I'm trying to solve is about when a request fails (because if network things) and we can't expect any status codes there. 

I'll need to update the PR.

Reply:
Then the event definition also need to be reconsidered. Status code isn't really applicable in this case.

Reply:
Hey @huoyaoyuan!
After some investigations I decided to keep only the duration and error message.